test/aionIntegrationTest.js

```diff
@@ -41,9 +44,89 @@ describe("Aion flavoured RLP integration", () => {
     txArray.push(new AionLong(new BN(2000000)));
     txArray.push(new AionLong(new BN("10000000000000")));
     txArray.push(new BN(1));
-    
+
     const output = RLP.encode(txArray);
     console.log(output.toString('hex'));
     assert.equal(output.toString('hex'), expectedOutput);
   });
-});
\ No newline at end of file
+
+  /*
+
+  ported from:
+  https://github.com/aionnetwork/aion/blob/master/modRlp/test/org/aion/rlp/RlpTestData.java
+
+  */
+
+  it("number 0", () => {
+    assertHex(0, "80");
+  });
+
+  it("number 1", () => {
+    assertHex(1, "01");
+  });
+
+  it("number 10", () => {
+    assertHex(10, "0a");
+  });
+
+  it("letter d", () => {
+    assertHex("d", "64");
+  });
+
+  it("string cat", () => {
+    assertHex("cat", "83636174");
+  });
+
+  it("string dog", () => {
+    assertHex("dog", "83646f67");
+  });
+
+  it("string array", () => {
+    assertHex(["cat", "dog"], "c88363617483646f67");
+  });
+
+  it("string array 2", () => {
+    assertHex(["dog", "god", "cat"], "cc83646f6783676f6483636174");
+  });
+
+  it("number 100", () => {
+    assertHex(100, "64");
+  });
+
+  it("number 1000", () => {
+    assertHex(1000, "8203e8");
+  });
+
+  it("BN 01", () => {
+    assertHex(
+      new BN("115792089237316195423570985008687907853269984665640564039457584007913129639935"),
+      "a0ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
+    );
+  });
+
+  it("BN 02", () => {
+    assertHex(
+      new BN("115792089237316195423570985008687907853269984665640564039457584007913129639936"),
+      "a1010000000000000000000000000000000000000000000000000000000000000000"
+    );
+  });
+
+  it("numbers and blank array", () => {
+    let input = [1, 2, []]
+    let output = "c30102c0"
+    assertHex(input, output);
+    // console.log(RLP.decode('0x' + output))
+    // assert.deepEqual(RLP.decode('0x' + output), input)
```

Comment:
Here I was trying to decode. The `Buffer` returned has the correct value but it did not deserialize back to a `number` type.

Reply:
I think that is the expected output but maybe you know better.